stdlib/public/runtime/Metadata.cpp

```diff
@@ -3655,6 +3655,19 @@ class ForeignMetadataCacheEntry
         state = PrivateMetadataState::NonTransitiveComplete;
       }
     } else {
+      if (candidate->getValueWitnesses() == nullptr) {
+#if SWIFT_OBJC_INTEROP
+        if (const auto *metadata = dyn_cast<ClassMetadata>(candidate)) {
+          const bool IsNativeObject =
+              (metadata->getFlags() & ClassFlags::UsesSwiftRefcounting);
+          candidate->setValueWitnesses(IsNativeObject
+                                           ? &VALUE_WITNESS_SYM(Bo)
+                                           : &VALUE_WITNESS_SYM(BO));
+        }
```

Comment:
I think you have to cast to `ForeignClassMetadata`, not `ClassMetadata`; foreign class metadata don't have these flags.  Also, you can just `cast`, not `dyn_cast`; this should be only the case where foreign type metadata don't have value witnesses.  Also, please add a comment explaining what this is doing.

Reply:
Sorry, Im not understanding something here.  What is the "other" side of the `#if` here?  The objc interop mode?  Or do you mean that we just assume that the VWT is BO?

Reply:
In non-ObjC interop, you should still assert that you're working with foreign-class metadata before setting the VWT.

Reply:
Actually, since `ForeignClassMetadata` does not have the flags, I think that it just isn't worth it - for now, I'll set it unconditionally to the unknown object on an ObjC-enabled runtime and the builtin object on a non-ObjC enabled runtime and assert before the condition.

Reply:
Sounds good.  But the compiler can't take advantage of it on Darwin until there's target-runtime logic.

Reply:
Yeah, its somewhat unfortunate, but I am leaving a FIXME there, and, I think that I shall open a SR for this.